### PR TITLE
feat(solana): update to 1.17.3 for installation

### PR DESCRIPTION
**Pull request by ifiokjr**
## Description of changes

Update to `1.17.3` for installation.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Please change the commit message so it follows the guidelines in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions.

**Comment by happysalada** [comment]
Did you try to build this ? The sha256 and the cargosha should both change as well. (If you try to build , it should give you the correct sha to replace )

File: pkgs/applications/blockchains/solana/default.nix [hunk 1]
```diff
@@ -12,36 +12,36 @@
...
     # XXX: Ensure `solana-genesis` is built LAST!
     # See https://github.com/solana-labs/solana/issues/5826
     "solana-genesis"
   ]
 }:
 let
-  version = "1.14.23";
+  version = "1.17.3";
   sha256 = "sha256-NUkkLzLNh8P7PFh/SVtd9JM18w3egDaaK80urGw1SSs=";
   cargoSha256 = "sha256-7t8Quh6T2MzJWEM5Y50CgCyFfx2ZJRAdCpZyyYvJrt4=";
```
Comment (eclairevoyant) [inline pkgs/applications/blockchains/solana/default.nix:46]:
```suggestion
  cargoHash = "";
```

File: pkgs/applications/blockchains/solana/default.nix [hunk 2]
```diff
@@ -12,36 +12,36 @@
...
 ] ++ [
     # XXX: Ensure `solana-genesis` is built LAST!
     # See https://github.com/solana-labs/solana/issues/5826
     "solana-genesis"
   ]
 }:
 let
-  version = "1.14.23";
+  version = "1.17.3";
   sha256 = "sha256-NUkkLzLNh8P7PFh/SVtd9JM18w3egDaaK80urGw1SSs=";
```
Comment (eclairevoyant) [inline pkgs/applications/blockchains/solana/default.nix:45]:
```suggestion
  hash = "";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
To clarify the previous comment: please initially set the hashes as below, then rebuild. The error will tell you the correct hashes.

**Comment by wegank** [comment]
Done in #301355.

**Pull request closed without merging**